moirae/models/ecm/utils.py

```diff
@@ -73,6 +73,8 @@ def get_value(self, soc: Union[Number, List, np.ndarray]) -> np.ndarray:
         if soc_batch_size > 1 and batch_size > 1:
             y = np.diag(y.squeeze())  # Match the SOC with the model its calling
         elif soc_batch_size == 1 and batch_size > 1:
+            if len(input_dims) > 1:
```

Comment:
Do you have a simple test which breaks before making this change? It'd be great to add one so that we ensure any future modifications (not that I have them in mind, it's just good to be prepared) don't revert to bad behavior.

Reply:
Yes, so the problem is that, in the `calculate_terminal_voltage` method inside the ECM, we are generally expecting arrays to be returned as 2D, with the first dimension being `batch_size`, and the other one, some parameterization-related dimension (in most cases, it should be 1). 

However, in this function, when the SOC is provided as a `(1,1)`-shaped array, and interpolation is needed, the returned vector is 3D, with shape `(batch_size,1,1)`. So, when we go forward to a next step that, for example, doesn't need interpolation, and returns an array of shape `(batch_size,1)`, numpy's broadcasting sums these together into an array of shape `('batch_size, batch_size,1)`, which cannot be properly squeezed. 

Here is an example of this: create a Thenevin model in which `R0` needs interpolation, but the resistor in the `RC` does not, but both are batched:
```python
# Define the values of R0 to use
r0_vals = np.array([5,4,6])

# Get A-SOH and set R0 values and interpolation style
asoh = ECMASOH.provide_template(has_C0=False, num_RC=1, R0=r0_vals)
asoh.r0.interpolation_style = 'quadratic'

# Mark R0 and RC as updatable
asoh.mark_updatable(name='r0.base_values')
asoh.mark_updatable(name='rc_elements.0.r.base_values')
# asoh_rint.mark_updatable(name='rc_elements.0.c.base_values')

# Now, create a batch of size 4
r0_batch = np.arange(12).reshape((4,3))
r_rc_batch = (1 + np.arange(4).reshape((4,-1)))
asoh.update_parameters(values=np.hstack((r0_batch, r_rc_batch)))

# Set up transient and input
transient = ECMTransientVector.provide_template(has_C0=False, num_RC=1, soc=1.0)
control = ECMInput(time=0., current=1.)

# Get terminal voltage
out = ECM().calculate_terminal_voltage(new_inputs=control,
                                       transient_state=transient,
                                       asoh=asoh)
```

The last line will raise a `pydantic` `ValidationError` from the `ScalarParameter` because the dimensions of the array are `(4, 4, 1)`:
```python
ValidationError: 1 validation error for ECMMeasurement
terminal_voltage
  Value error, Inconsistent dimensionality. Trying to store a 2 array as a scalar [type=value_error, input_value=array([[[ 5.7944439],
   ...
        [14.7944439]]]), input_type=ndarray]
```

Reply:
Can you reproduce that error with a simple test which uses only the SOC interpolation? I think I see the problem, but let's make sure we can express it in a simple form and add a test so that we can ensure it never becomes un-fixed.

Reply:
Yep, adding a test on the `test_components.py`